src/test/java/com/auth0/AuthenticationControllerTest.java

```diff
@@ -345,20 +352,156 @@ public void shouldProcessRequest() throws IdentityVerificationException {
         AuthenticationController controller = new AuthenticationController(requestProcessor);
 
         HttpServletRequest req = new MockHttpServletRequest();
-        controller.handle(req);
+        HttpServletResponse response = new MockHttpServletResponse();
 
-        verify(requestProcessor).process(req);
+        controller.handle(req, response);
+
+        verify(requestProcessor).process(req, response);
     }
 
     @Test
     public void shouldBuildAuthorizeUriWithRandomStateAndNonce() {
         RequestProcessor requestProcessor = mock(RequestProcessor.class);
         AuthenticationController controller = new AuthenticationController(requestProcessor);
 
-        HttpServletRequest req = new MockHttpServletRequest();
-        controller.buildAuthorizeUrl(req, "https://redirect.uri/here");
+        HttpServletRequest request = new MockHttpServletRequest();
+        HttpServletResponse response = new MockHttpServletResponse();
+
+        controller.buildAuthorizeUrl(request, response,"https://redirect.uri/here");
+
+        verify(requestProcessor).buildAuthorizeUrl(eq(request), eq(response), eq("https://redirect.uri/here"), anyString(), anyString());
+    }
+
+    @Test
+    public void shouldSetLaxCookiesAndNoLegacyCookieWhenCodeFlow() {
+        MockHttpServletResponse response = new MockHttpServletResponse();
+
+        AuthenticationController controller = AuthenticationController.newBuilder("domain", "clientId", "clientSecret")
+                .build();
+
+        controller.buildAuthorizeUrl(new MockHttpServletRequest(), response, "https://redirect.uri/here")
+                .withState("state")
```

Comment:
explicitly set that this is `code` flow

Reply:
Updated to explicitly set code response type on the controller.